tests/e2escenarios/core_beta_flow.go

```diff
@@ -0,0 +1,169 @@
+			Eventually(urlCreateSession).Should(gbytes.Say("odo push"))
+			helper.FileShouldContainSubstring(configFile, "Url:")
+			helper.FileShouldContainSubstring(configFile, "Port: 8080")
+
+			helper.CmdShouldPass(odo, append([]string{"push"}, extraArgs...)...)
+
+			dcSession := oc.GetComponentDC("mycomponent", "myapp", project)
+			Eventually(dcSession).Should(gbytes.Say("app.kubernetes.io/component-name: mycomponent"))
+			Eventually(dcSession).Should(gbytes.Say("app.kubernetes.io/component-source-type: local"))
+			Eventually(dcSession).Should(gbytes.Say("app.kubernetes.io/component-type: java"))
+			Eventually(dcSession).Should(gbytes.Say("app.kubernetes.io/name: myapp"))
+			Eventually(dcSession).Should(gbytes.Say("name: mycomponent-myapp"))
+			// DC should have env variable
+			// currently failing
+			Eventually(dcSession).Should(gbytes.Say("name: FOO"))
+			Eventually(dcSession).Should(gbytes.Say("value: bar"))
+
+			routeSession := oc.GetComponentRoutes("mycomponent", "myapp", project)
+			// check that route is pointing gto right port and component
+			Eventually(routeSession).Should(gbytes.Say("targetPort: 8080"))
+			Eventually(routeSession).Should(gbytes.Say("name: mycomponent-myapp"))
+			url := oc.GetFirstURL("mycomponent", "myapp", project)
+
+			helper.HttpWaitFor("http://"+url, "Hello World from Javalin!", 10, 5)
+		}
+
+		Context("when component is in the current directory", func() {
+
+			// we will be testing components that are created from the current directory
+			// switch to the clean context dir before each test
+			JustBeforeEach(func() {
+				originalDir = helper.Getwd()
+				helper.Chdir(context)
+			})
+
+			// go back to original directory after each test
+			JustAfterEach(func() {
+				helper.Chdir(originalDir)
+			})
+
+			It("'odo component' should fail if there already is .odo dir", func() {
+				helper.CmdShouldPass("odo", "component", "create", "nodejs")
+				helper.CmdShouldFail("odo", "component", "create", "nodejs")
+			})
+
+			It("'odo config' should fail if there is no .odo dir", func() {
+				helper.CmdShouldFail("odo", "config", "set", "memory", "2Gi")
+			})
+
+			It("create local java component and push code", func() {
+				helper.CopyExample(filepath.Join("source", "openjdk"), context)
+				TestBasicCreateConfigPush()
+			})
+
+		})
+
+		Context("when --context flag is used", func() {
+			// Set active project for each test spec
+			JustBeforeEach(func() {
+				originalProject = oc.GetCurrentProject()
+				oc.SwitchProject(project)
+			})
+			// go back to original project after each test
+			JustAfterEach(func() {
+				oc.SwitchProject(originalProject)
+			})
+
+			It("odo component should fail if there already is .odo dir", func() {
+				helper.CmdShouldPass("odo", "component", "create", "nodejs", "--context", context)
+				helper.CmdShouldFail("odo", "component", "create", "nodejs", "--context", context)
+			})
+
+			It("odo config should fail if there is no .odo dir", func() {
```

Comment:
this test should be put before previous test because previous test will create `.odo` dir